### PR TITLE
#2095 issue ticketing system

### LINKED ISSUE #2095: Digital issue resolution ticketing system
Background
Currently, users share digital platform issues with the Digital team through a Slack channel, but the system does not allow easy tracking, assigning and monitoring of issue resolution, metrics, etc.

User Story/Stories
As a user, I would like to be able to quickly share an issue on the digital platform with the digital team so that they can be resolved.

As the digital team, I would like to view all the pertinent information about an issue identified by a user so that it can be resolved without any additional correspondence.

### PR DESCRIPTION
**Pull request by drieJAC**
## What's included?

**SEE PERMISSIONS CHANGES SECTION BELOW!!**

**This is an initial prototype intended for an initial discussion on the features.**
The features are based on Tom's Miro: https://miro.com/app/board/uXjVMvZy_gs=/
Added a tab to the top right of the screen which will appear on all pages
When you click the tab a form will appear in a modal with some fields completed.
The user can specify how they would like to be contacted. If they choose 'email' then their email address will be prefilled.

Closes #2095 

## Who should test?
✅ Product owner

## How to test?

**UT On 18/10/2023**
Go to any page and click the feedback link in the top right hand corner.
View the fields that are present and give your feedback to the team.

- When raising an issue for a candidate display a warning saying "Please ensure you are on the relevant candidate application page before raising the issue"
- When raising an issue for a candidate include a prompt to ask "Have you checked if the candidate is using a CPS device?"
- When raising an issue for myself display a warning saying "Please ensure you are on the relevant page before raising the issue"

**Internal UT On 25/10/2023**

- Go to the preview url: https://jac-admin-develop--pr2117-feature-2095-issue-t-o8pguw8x.web.app/
- Ensure the 'Report An Issue' button is now in the bottom left hand corner of the screen and stays there as you scroll vertically
- Select an exercise
- Click the button to 'Report An Issue'  and fill in the form for 'Myself' ensuring you take a screenshot
- Submit the form and note the bug number in the popup
- (Dev Team Only) Go to the file storage area in firestore and ensure the file is under the bugReport/{bugReportId} directory
- Check that the bug has been reported in Zenhub's Platform Issues workspace: https://app.zenhub.com/workspaces/platform-issues-65310000d7eb371741bcfe5e/board

Screenshot 2023-11-01 at 15 15 16

- Check that the bug appears in Slack's platform-issues channel

Slack

## Risk - how likely is this to impact other areas?
🟢 No risk - this is a self-contained piece of work

## Additional context
See screenshot:

Screenshot 2023-08-29 at 12 54 03

## Related permissions
Have permissions been considered for this functionality?
- Permission changes required

Need to allow users to create zenhub issues with the following permission: canCreateZHIssues

---
PREVIEW:DEVELOP
_can be OFF, DEVELOP or STAGING_

### CODE REVIEW

**Comment by nickaddy** [comment]
@HalcyonJAC Thanks for putting together the prototype, Omar. We gathered some internal feedback at the sprint planning meetings and it's shared on [this Miro board](https://miro.com/app/board/uXjVMp_sMWo=/?share_link_id=566904512024).

**Review by tomlovesgithub — APPROVED** [review]

File: src/components/Feedback/FeedbackForm.vue [hunk 1]
```diff
@@ -0,0 +1,305 @@
...
+          {{ criticalityType }}
+        </option>
+      </Select>
+      <FileUpload
+        id="screenshot-file"
+        ref="screenshot-file"
+        v-model="formData.screenshotFileName"
+        name="screenshot"
+        :path="screenshotUploadPath"
+        types="fileTypes"
```
Comment (HalcyonJAC) [inline src/components/Feedback/FeedbackForm.vue:60]: I guess here it should use `:types="fileTypes"` to pass the variable into the component.

**Review by HalcyonJAC — CHANGES_REQUESTED** [review]

**Review by HalcyonJAC — APPROVED** [review]
The "Raise A Support Issue With This Page" button will overlap other buttons on a small screen. Not sure if we need to fix it in this PR as it's a prototype now so I just pointed this issue out.

image

**Comment by nickaddy** [comment]
User feedback was positive, but issues with uploading files. Could we consider a V2 that permits users submitting a screengrab as they currently do via Slack? If this is going to be very complicated, we might just drop the idea as all the other info we have captured may replace the screengrab?

**Comment by nickaddy** [comment]
@drieJAC @warrensearle Other comment around button placement overlapping with other text on screen.

**Comment by nickaddy** [comment]
@drieJAC @warrensearle Warren and I realised this needs a bit of a tweak earlier - to cater for cases where a user is raising an issue for an applicant. Hence we may need a multi-modal/adaptive modal approach. The first question would be: Are you raising this issue for yourself or an applicant? 

Option 1: Myself > present the options in your current modal
Option 2: Applicant > present the questions below

Name of applicant, Browser type, etc, the usual questions in addition to the 2/3 we have on the current modal, probably without the screenshot option as they won't be able to do one on behalf of the applicant. Perhaps we should add guidance to the user to ensure they are on the relevant exercise page before raising the issue?

Does that make sense?

Longer term, this feature may be rolled out to the apply side, but in the the meantime, users need a mechanism for raising candidate issues.

**Comment by drieJAC** [comment]
Ok @nickaddy @warrensearle those changes have been done

**Comment by nickaddy** [comment]
@drieJAC @warrensearle Love what you've done with this Omar :-) I have a couple of comments:

- Notice you've changed the format of the raise an issue button (in relation to display overlap issues?), if we're going with an icon, could it be sthg a bit more obvious like, e.g. a red exclamation mark? Whatever we choose, I suppose SETs will soon get familiar with.
- On the raising for someone else form, I wonder if we should add sthg about issues on CPS devices or maybe that is sthg that is communicated separately to SETs as a first step? Open to suggestions...

Anyway, let's put it back to UT next week :-)

**Comment by warrensearle** [comment]
Looks like the message to "Please ensure you are on the relevant page..." is on the wrong page. It should go with the capture screenshot functionality shouldn't it?

image

**Comment by warrensearle** [comment]
When gathering data from the applicant we should ask for the URL of the page the applicant has a problem with.

Perhaps we should also refer to them as "candidate" rather than "applicant". Both are correct but the tendency is to call them candidates

**Comment by nickaddy** [comment]
@warrensearle This was so we could capture the exercise details that the candidate was involved in, rather than having to ask them to submit a url. It could also be mentioned on the 'myself' form.

**Comment by warrensearle** [comment]
Ah right so we need to make the following changes:

- [ ] Change "Applicant" to "Candidate"
- [ ] Change the warning when raising an issue for a candidate application to "Please ensure you are on the relevant candidate application page before raising the issue"
- [ ] When raising an issue for a candidate include a prompt to ask "Have you checked if the candidate is using a CPS device?"
- [ ] Include a warning when raising an issue for myself to say "Please ensure you are on the relevant page before raising the issue"

**Review by warrensearle — CHANGES_REQUESTED** [review]
Please see comments

**Comment by drieJAC** [comment]
Add a link to the screenshot to the issue in Zenhub and a link to the Zenhub issue in Slack

**Review by nickaddy — CHANGES_REQUESTED** [review]
Partly successful: ticket created in ZenHub and data passed to Firestore collection, but content of user-completed fields was truncated (I specifically added a lot of text to test this). Screenshot also needs to be linked to in ZH ticket (Omar aware.)

**Comment by nickaddy** [comment]
@drieJAC I should have added that it all works for candidates too

**Comment by nickaddy** [comment]
@drieJAC to discuss at this week's Devs meeting which additional data, e.g. OS, browser, user email should be passed to the ZenHub ticket.

Additional request: can 'myself' issues be auto-allocated a 'User' label in ZH and 'candidate' issues be allocated a 'Candidate' label in ZH.

**Comment by nickaddy** [comment]
@drieJAC What did you expect to happen user input needs to be passed to ZH/Slack

**Comment by nickaddy** [comment]
@drieJAC Can we automate a label 'apply' or 'admin' to the ticket according to whether ticket is candidate or user related?

**Comment by nickaddy** [comment]
@drieJAC When ticket assigned in ZH, send notification to developer assigned in Slack channel.

**Comment by nickaddy** [comment]
@drieJAC when issue is resolved, user is notified.

**Comment by nickaddy** [comment]
@drieJAC @warrensearle I have created a [messages section](https://miro.com/app/board/uXjVMgsSPkg=/?moveToWidget=3458764568633829550&cot=14) on the Miro board.

**Comment by nickaddy** [comment]
Otherwise, Omar & I agreed it is OK for additional conversation to take place in digital issues Slack channel thread if required with developer pasting anything pertinent into the ZH ticket.

Suggest a live UT session later in the month, where users raise issues and we triage them live.

**Comment by nickaddy** [comment]
@drieJAC @warrensearle Just had a thought: if we are passing the assignee back to the reporter, we need to either make aliases for our usernames in GH/ZH or change our usernames to reflect at least our first names. Thoughts?

**Comment by nickaddy** [comment]
@drieJAC @warrensearle Sorry about the multiple messages - I was hoping we would catch up today but with Warren still in the Canaries, that's not going to happen. We are going to need to link this board to a GH repo: 1) for automations, 2) for metrics.

**Comment by drieJAC** [comment]
Investigate...

- Add label to zenhub issue.
- Send emails on change of status.

**Comment by nickaddy** [comment]
@warrensearle Modal is generating an error today - not sure if Omar made changes yesterday - oddly, it does actually send to Slack and ZH, but not all of the data.
[Private Zenhub Image](https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBeGYyQVE9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--7612a1b9d16ca549ffb5f6ae6cdc1c39b54d25dc/image.png)

**Pull request merged**